pkg/client/factory.go

```diff
@@ -137,6 +143,21 @@ func (f *factory) DynamicClient() (dynamic.Interface, error) {
 	return dynamicClient, nil
 }
 
+func (f *factory) KubebuilderClient() (kbclient.Client, error) {
```

Comment:
`VeleroV1APIClient` seems like a better name for this function. WDYT?

Reply:
I'm not sure. Looking at the other clients, none has version in the name:
![image](https://user-images.githubusercontent.com/16508/85134006-a2cf8d80-b1f0-11ea-9ed7-c20079cb595c.png)

`Client` would be the best, but it's already in use so we need something to differentiate this new kubebuilder client.

@nrb do you have any opinion on this?

Reply:
Looking through what's returned, `VeleroV1APIClient` doesn't quite make sense to me, since we get https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client?tab=doc#Client, which is a generic client that can interact with anything in the Kubernetes API server, so it's not really just for Velero V1; we just happen to be using it for that now. Really, we could register any API group/scheme to it and use this client to get them.

In the current file, `Client` returns https://pkg.go.dev/github.com/vmware-tanzu/velero@v1.4.0/pkg/generated/clientset/versioned?tab=doc#Interface, which has the `VeleroV1()` method, which makes it a typed and specific client.

I'm leaning towards keeping the `KubeBuilderClient` name for now, then when we complete the transition, renaming it to `KubeClient`, which would replace the existing `KubeClient` for API server interactions. That's a much further-looking plan, though.

Thoughts?